### PR TITLE
Implements data slicing

### PR DESCRIPTION
**Pull request by mpetruska**
Simplified slice implementation, removed gap handling.

### CODE REVIEW

**Comment by mpetruska** [comment]
Contains enhancements related to https://github.com/eigengo/lift/issues/74.
Please review.

File: ios/Lift/Device/SensorData.swift [hunk 1]
```diff
@@ -68,6 +68,17 @@ struct TimeRange : Equatable {
     func within(that: TimeRange, maximumGap gap: CFTimeInterval) -> Bool {
         return (start - gap) >= that.start && (end + gap) <= that.end
     }
+    
+    ///
+    /// Indicates whether the given value is contained in thsi range.
```
Comment (carlpulley) [inline ios/Lift/Device/SensorData.swift]: `thsi` -> `this`

**Comment by carlpulley** [comment]
LGTM, :+1:

**Comment by mpetruska** [comment]
Typo fixed, thanks Carl.

File: ios/Lift/LiveSessionSensorDataGroupController.swift [hunk 2]
```diff
@@ -79,11 +96,17 @@ class SensorDataGroupView : UIView {
                     CGContextStrokeRect(ctx, frame)
                 }
                 y += height + padding
+                if i == 2 && lastEnd <= 0.01 {     // !!!!!!!!!
```
Comment (zapletal-martin) [inline ios/Lift/LiveSessionSensorDataGroupController.swift]: What does this bit do?

Comment (mpetruska) [inline ios/Lift/LiveSessionSensorDataGroupController.swift]: Ah, was using that for debugging, forgot to remove.
Also found something that should be fixed, please stand by.

File: ios/Lift/Device/SensorData+Time.swift [hunk 3]
```diff
@@ -55,4 +55,16 @@ extension SensorDataArray {
         }
     }
     
+    var longestAvailablePeriod: CFTimeInterval? {
```
Comment (janm399) [inline ios/Lift/Device/SensorData+Time.swift]: Needs comments

File: ios/Lift/Device/SensorData.swift [hunk 4]
```diff
@@ -68,6 +68,17 @@ struct TimeRange : Equatable {
...
     }
+    
+    ///
+    /// Indicates whether the given value is contained in this range.
+    ///
+    func contains(value: CFAbsoluteTime) -> Bool {
+        return (start <= value) && (value <= end)
+    }
+    
+    func intersects(other: TimeRange) -> Bool {
```
Comment (janm399) [inline ios/Lift/Device/SensorData.swift]: Needs docs

File: pebble/core/main/gfs.c [hunk 5]
```diff
@@ -25,7 +25,7 @@ void gfs_write_header() {
     struct gfs_header *h = (struct gfs_header *) gfs_context.buffer;
     h->type = GFS_HEADER_TYPE;
     h->count = 0;
-    h->sample_size = 5;
+    h->sample_size = sizeof(struct gfs_packed_accel_data);
```
Comment (janm399) [inline pebble/core/main/gfs.c]: :)

**Comment by janm399** [comment]
It needs docs throughout the newly added code, otherwise, I think we're good to go.

File: ios/Lift/Device/NSMutableData+Encoding.swift [hunk 6]
```diff
@@ -28,7 +28,7 @@ extension NSMutableData {
     }
     
     /// Appends the given bytes to this data
-    private func appendBytes(value: [UInt8]) -> Void {
+    func appendBytes(value: [UInt8]) -> Void {
```
Comment (janm399) [inline ios/Lift/Device/NSMutableData+Encoding.swift]: If you look at he other docs, the style is
```swift
///
/// Computes result something using the elements in ``x``
///
func foo(x: [X]) -> R {
}
```

You should follow it.

**Pull request merged**